### PR TITLE
PR auto deploy

### LINKED ISSUE #771: Pull Requests don't have preview Links
We need an auto-deploy mechanism for the pull requests made to the repo.

### LINKED ISSUE #772: .gitignore missing
there should be a .gitignore file to omit file changes

### LINKED ISSUE #773: Travis Outdated
Travis is not configured to 
- cache node modules
- use latest node version
- documented (via inline comments)

### PR DESCRIPTION
**Pull request by YashKumarVerma**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a description of change below.
- [x] Squashed Commits

### Changes done in this Pull Request
- travis.yml updated to call deploy.sh
- travis.yml updated to cache node modules to speed up deploys
- .gitignore added to ignore folders created during npm commands
- deploy.sh created to deploy every pr to a live site

### ISSUES
- Fixes #771 Pull Requests don't have preview Links 
- Fixes #772  .gitignore missing
- Fixes #773 Travis Outdated

### ATTENTION
***SUGGEST CHANGES TO ACCOMODATE HTML SYNTAX CHECK WITH NODE_MODULES***

### CODE REVIEW

**Comment by YashKumarVerma** [comment]
@niccokunzmann @abishekvashok 
please help to find a way to ignore ***node_modules*** from js syntax check.

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
```
Comment (niccokunzmann) [inline deploy.sh]: this should be in the scripts folder

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log("PUT YOUR TESTS HERE");
```
Comment (niccokunzmann) [inline test/test.js]: I do not think we need this file.

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+    - node_modules
 
+#quality checks
 script:
   - scripts/check-javascript-syntax-errors.sh
   - scripts/check-html-syntax-errors.sh
   
+# deploy pr 
+after_success:
+- bash ./deploy.sh
```
Comment (niccokunzmann) [inline .travis.yml]: You can mark the file executable and use scripts/deploy.sh

**Comment by YashKumarVerma** [comment]
I'll go the changes once I'm back from classes.

If someone else wants to, throw a pr to my repo

https://stackoverflow.com/questions/42154912/permission-denied-for-build-sh-file

**Comment by niccokunzmann** [comment]
Tests are running. I will merge it.

**Pull request merged**